src/quickgui/qgsquickpositionkit.cpp

```diff
@@ -0,0 +1,311 @@
+/***************************************************************************
+  qgsquickpositionkit.cpp
+  --------------------------------------
+  Date                 : Dec. 2017
+  Copyright            : (C) 2017 Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgis.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+
+#include "qgsquickpositionkit.h"
+#include "qgsquickutils.h"
+#include "qgsquicksimulatedpositionsource.h"
+
+QgsQuickPositionKit::QgsQuickPositionKit( QObject *parent )
+  : QObject( parent )
+  , mAccuracy( -1 )
+  , mScreenAccuracy( 2 )
+  , mAccuracyUnits( QgsUnitTypes::toAbbreviatedString( QgsUnitTypes::DistanceMeters ) )
+  , mDirection( -1 )
+  , mHasPosition( false )
+  , mIsSimulated( false )
+{
+  connect( this, &QgsQuickPositionKit::simulatePositionLongLatRadChanged, this, &QgsQuickPositionKit::onSimulatePositionLongLatRadChanged );
+  connect( this, &QgsQuickPositionKit::positionChanged, this, &QgsQuickPositionKit::updateScreenPosition );
+  connect( this, &QgsQuickPositionKit::positionChanged, this, &QgsQuickPositionKit::updateProjectedPosition );
+
+  use_gps_location();
+}
+
+QGeoPositionInfoSource  *QgsQuickPositionKit::gpsSource()
+{
+  // this should give us "true" position source
+  // on Linux it comes from Geoclue library
+  QGeoPositionInfoSource *source = QGeoPositionInfoSource::createDefaultSource( this );
```

Comment:
`source` could be a `unique_ptr` in this case to avoid the manual `delete` when an error happens.